# --- T2-COPYRIGHT-NOTE-BEGIN ---
# T2 SDE: package/*/xf86-video-cirrus/hotfix-laguna-crash.patch
# Copyright (C) 2024 The T2 SDE Project
# 
# This Copyright note is generated by scripts/Create-CopyPatch,
# more information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License version 2 as used by the T2 SDE.
# --- T2-COPYRIGHT-NOTE-END ---

--- xf86-video-cirrus-1.6.0/src/lg_driver.c.vanilla	2024-12-06 23:59:16.633284524 +0100
+++ xf86-video-cirrus-1.6.0/src/lg_driver.c	2024-12-07 00:06:01.336615266 +0100
@@ -475,7 +475,7 @@
      * The first thing we should figure out is the depth, bpp, etc.
      * We support both 24 bpp and 32 bpp layouts, so indicate that.
      */
-    if (!xf86SetDepthBpp(pScrn, 0, 0, 0, Support24bppFb |
+    if (!xf86SetDepthBpp(pScrn, 16, 16, 16, Support24bppFb |
                                             Support32bppFb |
                                             SupportConvert32to24 |
                                             PreferConvert32to24)) {
@@ -1879,7 +1879,8 @@
     /*
      * Where's the pointer?
      */
-    miPointerGetPosition(inputInfo.pointer, &cursorX, &cursorY);
+    //miPointerGetPosition(inputInfo.pointer, &cursorX, &cursorY);
+    cursorX = cursorY = 0;
 
     /*
      * Where's the middle of the screen?  We want to eventually know
